src/NUnitFramework/framework/Internal/Builders/DatapointProvider.cs

```diff
@@ -132,6 +132,13 @@ public System.Collections.IEnumerable GetDataFor(IParameterInfo parameter)
 
             if (datapoints.Count == 0)
             {
+                var underlyingParameterType = Nullable.GetUnderlyingType(parameterType);
+                if (underlyingParameterType != null)
+                {
+                    datapoints.Add(null);
+                    parameterType = underlyingParameterType;
+                }
+
```

Comment:
This will add a null for any underlying type that is not supplied, not just bool and enum.